Add Fusion offline authentication article (#364)
vaadin/docs: articles/fusion/pwa/offline-authentication-check.asciidoc

Take the following security aspects into account when you do authentication checks while the user is offline

== Authenticity

> Do you need to protect the local user's data on the device from whoever else may have access to the same device? If that is a requirement, then you will end up adding a `passcode` in addition to the regular online login (can be replaced with a fingerprint/face id auth through the WebAuthn API on the devices that support those). This is what many mobile bank apps do.

Reviewer: ```suggestion
Do you need to protect the local user's data on the device from whoever else may have access to the same device?
If that is a requirement, you need to add a `passcode` in addition to the regular online login (or use fingerprint/face ID authentication through the WebAuthn API on the devices that support those).
This is what many mobile bank apps do.
```

It is to encrypt any user data that is cached locally for offline use.
If you do go that route, most likely you will end up having a helper library that you can call to check if the user is logged in or not.

Arguably, not many apps would need to securely verify the user authenticity offline, and most can check if an online login happened in the not-so-distant past.

 - For that the simplest approach would be setting a local storage flag after a user has successfully logged in online (and clear it when the user logs out).
During offline navigation, the user auth then would be checked from the local storage.

> The <<../security/authentication-offline.asciidoc#, Offline Support for Authentication>> article describes how to do that.

Reviewer: ```suggestion
The <<../security/authentication-offline#, Offline Support for Authentication>> article describes how to do that.
```
